src/main/java/com/google/devtools/build/lib/remote/AbstractActionInputPrefetcher.java

```diff
@@ -291,30 +297,38 @@ public ListenableFuture<Void> prefetchFiles(
     // it must still synchronize on the output permissions having been set.
     Set<Path> dirsWithOutputPermissions = Sets.newConcurrentHashSet();
 
-    Completable prefetch =
-        mergeBulkTransfer(
-                Flowable.fromIterable(files)
-                    .flatMapSingle(
-                        input ->
-                            prefetchFile(
-                                action,
-                                dirsWithOutputPermissions,
-                                metadataSupplier,
-                                input,
-                                priority)))
-            .doOnComplete(
-                // Set output permissions on tree artifact subdirectories, matching the behavior of
-                // SkyframeActionExecutor#checkOutputs for artifacts produced by local actions.
-                () -> {
-                  for (Path dir : dirsWithOutputPermissions) {
-                    directoryTracker.setOutputPermissions(dir);
-                  }
-                });
-
-    return toListenableFuture(prefetch);
+    List<ListenableFuture<Void>> transfers = new ArrayList<>(files.size());
```

Comment:
Add a note here explaining why we're using plain futures instead of RxJava.

Reply:
Done.